source/instructors/authoring/assessments/auto-grade-scripts.rst

```diff
@@ -75,35 +75,65 @@ You can implement participation grading using assignment level scripts.
-    import json
+    .. code-tab:: bash
 
-    # import grade submit function
-    import sys
-    sys.path.append('/usr/share/codio/assessments')
-    from lib.grade import send_grade
+            #!/bin/bash
+
+            # save json based passed environment
+            echo $CODIO_AUTOGRADE_ENV > save.json
+
+            ANSWERED=$(jq -r '.assessments.stats.answered' save.json)
+            TOTAL=$(jq -r '.assessments.stats.total' save.json)
+
+            GRADE=$(($ANSWERED * 100 / $TOTAL))
 
-    env = os.environ.get('CODIO_AUTOGRADE_ENV')
+            FEEDBACK=""
 
-    parsed = json.loads(env)
+            if [ $TOTAL -eq $ANSWERED ]; then
+              FEEDBACK="✅ You answered all the questions and got full points on this assignment"
+            else
+              FEEDBACK="❌ You did not answer all the questions and therefore only received points for the questions you answered"
+            fi
 
-    answered = parsed['assessments']['stats']['answered']
+            curl --retry 3 -s "$CODIO_AUTOGRADE_V2_URL" -d grade=$GRADE -d format=md -d feedback="$FEEDBACK"
 
-    total=parsed['assessments']['stats']['total']
+    .. code-tab:: python 
 
-    grade=answered*100/total
-    res = send_grade(int(round(grade)))
-    exit( 0 if res else 1)
+            import os
+            import json
+
+            # import grade submit function
+            import sys
+            sys.path.append('/usr/share/codio/assessments')
+            from lib.grade import send_grade_v2, FORMAT_V2_MD, FORMAT_V2_HTML, FORMAT_V2_TXT
+
+            feedback=""
+            env = os.environ.get('CODIO_AUTOGRADE_ENV')
+            parsed = json.loads(env)
+            answered = parsed['assessments']['stats']['answered']
+            total=parsed['assessments']['stats']['total']
+
+            grade=answered*100/total
+
+            if total==answered:
+              feedback+="✅ You answered all the questions and got full points on this assignment"
+            else:
+              feedback+="❌ You did not answer all the questions and therefore only received points for the questions you answered"
+
+            res = send_grade_v2(grade, feedback, FORMAT_V2_MD)
+
+            exit( 0 if res else 1)
 
 
 - Add the file to **Education>Test Autograde Script**. If your file is not a bash script or other type of file that runs independently, you will need to specify the program that will run it, for example "python3 autograde.py".
 - Note: The JSON is not updated until the assignment is marked as complete. If you are testing values from inside the assignment - you will not see the updated values.
 - Make sure to **Publish** the assignment.
 - In the course assignment settings :ref:`Grade Weights <grade-weights>` section, enable **Script Grading** set **Set custom script path** to that file and disable **Assessments Grading**.
+- If you are using the Bash version you need to install the utility "jq" to your stack (see :ref:`create a new stack <create-stack>`).
```

Comment:
I would suggest moving the installation of dependencies earlier on the page, as it may trick the user into believing it is already installed.

Reply:
Good point! I have moved the information above the examples and committed.